### ISSUE TITLE
`torchmetrics.Accuracy` is too slow than normal pytorch code

### ISSUE THREAD

**Issue by marsggbo** [PyTorchLightning/metrics#485]
### To Reproduce

```python
import torch
import torchmetrics
x = torch.rand(256,10)
y = torch.randint(10,(256,))

def MyAccuracy(output, target, topk=(1,)):
    """ Computes the precision@k for the specified values of k """
    maxk = max(topk)
    batch_size = target.size(0)

    _, pred = output.topk(maxk, 1, True, True)
    pred = pred.t()
    correct = pred.eq(target.reshape(1, -1).expand_as(pred))

    res = []
    for k in topk:
        correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
        res.append(torch.tensor(correct_k.mul_(100.0 / batch_size)))
    if len(res) == 1:
        return res[-1]
    return res
```

timing

```python
import timeit
def f1():
    MyAccuracy(x,y)
def f2():
    torchmetrics.Accuracy()(x,y)
def f3():
    torchmetrics.functional.accuracy(x,y)

timeit.timeit(f1, number=1000)
timeit.timeit(f2, number=1000)
timeit.timeit(f3, number=1000)
```

The result is on CPU

![image](https://user-images.githubusercontent.com/13477956/131214030-13a45369-6a10-481d-bce3-d63c27317d31.png)

### Environment

- PyTorch Version (e.g., 1.0):
- OS (e.g., Linux):
- How you installed PyTorch (`conda`, `pip`, source):
- Build command you used (if compiling from source):
- Python version:
- CUDA/cuDNN version:
- GPU models and configuration:
- Any other relevant information:

**Comment by Borda** [comment]
@marsggbo that is a good finding, interested in sending faster implementation?
I would say that the overhead comes with various validation checks and update/compute processes under the hood...
cc: @PyTorchLightning/core-metrics

**Comment by puhuk** [comment]
@Borda I send PR #609 regarding this issue, could you review this :)
